doc/src/sgml/installation-meson.sgml

```diff
@@ -0,0 +1,1526 @@
+       <listitem>
+        <para>
+         Build with support for
+         the <productname>ICU</productname><indexterm><primary>ICU</primary></indexterm>
+         library, enabling use of ICU collation
+         features<phrase condition="standalone-ignore"> (see
+         <xref linkend="collation"/>)</phrase>.
+         This requires the <productname>ICU4C</productname> package
+         to be installed.  The minimum required version
+         of <productname>ICU4C</productname> is currently 4.2.
+         It defaults to auto, meaning that it will be enabled automatically if the
+         required packages are found.
+        </para>
+
+        <para>
+         By default,
+         <productname>pkg-config</productname><indexterm><primary>pkg-config</primary></indexterm>
+         will be used to find the required compilation options.  This is
+         supported for <productname>ICU4C</productname> version 4.6 and later.
+         For older versions, or if <productname>pkg-config</productname> is
+         not available, the variables <envar>ICU_CFLAGS</envar>
+         and <envar>ICU_LIBS</envar> can be specified
+         to <filename>configure</filename>, like in this example:
+<programlisting>
+meson configure ... -Dicu ICU_CFLAGS='-I/some/where/include' ICU_LIBS='-L/some/where/lib -licui18n -licuuc -licudata'
+</programlisting>
+         (If <productname>ICU4C</productname> is in the default search path
+         for the compiler, then you still need to specify nonempty strings in
+         order to avoid use of <productname>pkg-config</productname>, for
+         example, <literal>ICU_CFLAGS=' '</literal>.)
+        </para>
+       </listitem>
+      </varlistentry>
+
+      <varlistentry id="configure-with-llvm-meson">
+       <term><option>-Dllvm=<replaceable>enabled/disabled/auto</replaceable></option></term>
+       <listitem>
+        <para>
+         Build with support for <productname>LLVM</productname> based
+         <acronym>JIT</acronym> compilation<phrase
+         condition="standalone-ignore"> (see <xref
+         linkend="jit"/>)</phrase>.  This
+         requires the <productname>LLVM</productname> library to be installed.
+         The minimum required version of <productname>LLVM</productname> is
+         currently 3.9. It is set to disabled by default.
+        </para>
+        <para>
+         <command>llvm-config</command><indexterm><primary>llvm-config</primary></indexterm>
+         will be used to find the required compilation options.
+         <command>llvm-config</command>, and then
+         <command>llvm-config-$major-$minor</command> for all supported
+         versions, will be searched for in your <envar>PATH</envar>.  If
+         that would not yield the desired program,
+         use <envar>LLVM_CONFIG</envar> to specify a path to the
+         correct <command>llvm-config</command>. For example
+<programlisting>
+meson configure ... --with-llvm LLVM_CONFIG='/path/to/llvm/bin/llvm-config'
```

Comment:
I did not and I should have modified it better as this anyway isn't likely to work. Will it work with the below command?

`LLVM_CONFIG=''/path/to/llvm/bin/llvm-config' meson configure ... --with-llvm`

Reply:
As discussed on IM: No, that won't currently help.

Reply:
Meson has some programs it looks up from the environment, see `mesonbuild/envconfig.py` for details.

It will look up things like AR, RANLIB, CC, PKG_CONFIG, CMAKE, VAPIGEN.

It won't look up LLVM_CONFIG, because it's not in that list.

Why is it not in that list? I don't know. Perhaps someone would like to submit a pull request adding it, hint hint. ;)

Reply:
@eli-schwartz : Submitted a first draft PR adding it here: https://github.com/mesonbuild/meson/pull/10757 :)

Any tips on what kind of tests would be appropriate?

Reply:
It's using a standard layer in common with other build tools -- and other config-tool dependencies as well -- so I don't think you need to test that it works for llvm specifically.

Thanks! And comments added.